tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/TieredStoreProvider.java

```diff
@@ -19,7 +19,7 @@
 import java.nio.ByteBuffer;
 import java.util.concurrent.CompletableFuture;
 
-public interface TieredStoreBackendProvider {
```

Comment:
Could we use Java SPI mechanism to get the implementation?

Reply:
The SPI mechanism is not applicable in this case. We want users to build their own provider by extending TieredFileSegment, which is an abstract class, not an interface. And tiered storage will create multiple file segment instances in multiple threads. It's more suitable to use the class name to initialize the file segment objects directly.